p2p/transport/websocket/websocket_test.go

```diff
@@ -548,3 +548,71 @@ func TestResolveMultiaddr(t *testing.T) {
 		})
 	}
 }
+
+func TestListenerResusePort(t *testing.T) {
+	laddr := ma.StringCast("/ip4/127.0.0.1/tcp/5002/ws")
+
+	var opts []Option
+	opts = append(opts, EnableReuseport())
+	_, u := newUpgrader(t)
+	tpt, err := New(u, &network.NullResourceManager{}, opts...)
+	require.NoError(t, err)
+	c := make(chan int)
+
+	for i := 0; i < 2; i++ {
+		go func(index int, ch chan int) {
+			l, err := tpt.maListen(laddr)
+			if err != nil {
+				fmt.Println("Failed to listen on websocket due to error ", err)
+			}
+			require.NoError(t, err)
+			require.Equal(t, lastComponent(t, l.Multiaddr()), wsComponent)
+			defer l.Close()
+			c <- index
+			/* Looping 4 times to ensure all 4 connections are handled.
+			   Noticed that sometimes the distribution of connections was
+			   not clearly load-balanced leading to more than 2 being delivered to 1 listener. */
+			for j := 0; j < 4; j++ {
+				conn, err := l.Accept()
+				if err != nil {
+					fmt.Println("Routine-", index, " Failed accepting connection due to error ", err)
+				}
+				require.NoError(t, err)
+				defer conn.Close()
+				buf := make([]byte, 6)
+				n, err := conn.Read(buf)
+				if n != 6 {
+					t.Errorf("read %d bytes, expected 2", n)
+				}
+				require.NoError(t, err)
+				c <- j
+			}
+		}(i, c)
+	}
+	for i := 0; i < 2; i++ {
+		<-c
+	}
+	for i := 0; i < 4; i++ {
+		go func(index int, ch chan int) {
+			c, err := tpt.maDial(context.Background(), laddr)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			require.NoError(t, err)
+			defer c.Close()
+			msg := fmt.Sprintf("Hello%d", i)
```

Comment:
you probably want `index` here. Thanks linter!

Reply:
Ah..my bad. Pushing an update.